File: src/model/Restriction.ts
export class Restriction{
  private _idRestriction !: number;
  private _lancerEmail !: string;
  private _debut !: Date;
  private _fin !: Date;
  private _raison !: string;
  private _enExecution !: boolean;


  constructor(idRestriction: number, lancerEmail: string, debut: Date, fin: Date, raison: string, enExecution: boolean) {
    this._idRestriction = idRestriction;
    this._lancerEmail = lancerEmail;
    this._debut = debut;
    this._fin = fin;
    this._raison = raison;
    this._enExecution = enExecution;
  }


  get idRestriction(): number {
    return this._idRestriction;
  }

  set idRestriction(value: number) {
    this._idRestriction = value;
  }

  get lancerEmail(): string {
    return this._lancerEmail;
  }

  set lancerEmail(value: string) {
    this._lancerEmail = value;
  }

  get debut(): Date {
    return this._debut;
  }

  set debut(value: Date) {
    this._debut = value;
  }

  get fin(): Date {
    return this._fin;
  }

  set fin(value: Date) {
    this._fin = value;
  }

  get raison(): string {
    return this._raison;
  }

  set raison(value: string) {
    this._raison = value;
  }

  get enExecution(): boolean {
    return this._enExecution;
  }

  set enExecution(value: boolean) {
    this._enExecution = value;
  }
}

File: lancersBackEnd/model/Restriction.py
from datetime import datetime

class Restriction:
    def __init__(self, idRestriction: int, lancerEmail: str, debut: datetime, fin: datetime, raison: str, enExecution: bool):
        self._idRestriction = idRestriction
        self._lancerEmail = lancerEmail
        self._debut = debut
        self._fin = fin
        self._raison = raison
        self._enExecution = enExecution

    @property
    def idRestriction(self) -> int:
        return self._idRestriction

    @idRestriction.setter
    def idRestriction(self, value: int):
        self._idRestriction = value

    @property
    def lancerEmail(self) -> str:
        return self._lancerEmail

    @lancerEmail.setter
    def lancerEmail(self, value: str):
        self._lancerEmail = value

    @property
    def debut(self) -> datetime:
        return self._debut

    @debut.setter
    def debut(self, value: datetime):
        self._debut = value

    @property
    def fin(self) -> datetime:
        return self._fin

    @fin.setter
    def fin(self, value: datetime):
        self._fin = value

    @property
    def raison(self) -> str:
        return self._raison

    @raison.setter
    def raison(self, value: str):
        self._raison = value

    @property
    def enExecution(self) -> bool:
        return self._enExecution

    @enExecution.setter
    def enExecution(self, value: bool):
        self._enExecution = value
